sql/mysqld.cc

```diff
@@ -5246,6 +5248,13 @@ static int init_server_components()
   error_handler_hook= my_message_sql;
   proc_info_hook= set_thd_stage_info;
 
+  /*
+    Print source revision hash, as one of the first lines, if not the
+    first in error log, for troubleshooting and debugging purposes
+  */
+  sql_print_information("Starting MariaDB %s source revision %s",
```

Comment:
It might be useful to standardize on showing this in the same format that the client uses, containing additional source/architecture info:

https://github.com/MariaDB/server/blob/010273268665746e188324e49730704c83aa8752/client/mysql.cc#L1717-L1724